### PR TITLE
Basic database constraints  + addition of requiring foreign keys.

### LINKED ISSUE #111: Foreign keys/indexes in database definitions
### What you would like to change/add
We currently have a mismatch of preferences when it comes to how our databases are scaffolded.

We have the privilege of using the MySQL/InnoDB combination by default in Laravel development, which actually automatically indexes foreign keys (for all versions to current since at least 5.5), making the work required incredibly small.

I propose that we enforce the definition of foreign keys on all relations, and additionally if the system is not MySQL based - the addition of indexes if the engine/SQL version requires separate definition. I would say this could be done on a case-by-case but it's best to blanket add them IMO, even if only to match the behaviour of 99% of our systems (MySQL/InnoDB).

### Why you would like to change/add this
I've had to deal with a few projects now grinding to a halt after 6 months of usage by clients because the original database/migrations didn't include the usage of foreign keys/indexes.

Commercially this is extremely bad for us and causes businesses to literally not be able to work with the software that's running their business, they get frustrated at us - and it causes unnecessary, unscheduled maintenance where we have to drop what we're doing to deal with it.

This would include the use of the `ON DELETE` and `ON UPDATE` triggers on foreign keys, but I feel that that issue is slightly more controversial so will be handled separately. This issue ONLY relates to the definition of the FKs themselves.

### Examples
```
$table->unsignedInteger('user_id');
$table->foreign('user_id')->references('id')->on('users'); //onDelete('cascade');
```

### Checklist
- [x] I have checked that this is not a duplicate issue.

### Sources
https://dev.mysql.com/doc/refman/5.5/en/constraint-foreign-key.html
https://stackoverflow.com/questions/304317/does-mysql-index-foreign-key-columns-automatically

### PR DESCRIPTION
**Pull request by wheatleyjj**
Closes #111. 

## Checklist

- [x] I have ensured that I have used relative links where appropriate.
- [x] I have proofread my documentation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,6 +16,7 @@ We work with a number of different software platforms. Each has their own set of
   * [📲 React Native](platforms/mobile/react-native/introduction.md)
 * 🖥 Desktop
   * [🖼 Windows apps](platforms/desktop/windows.md)
+* 🗂  [Databases](databases.md)
```
Comment (dextermb) [inline README.md]:
```suggestion
* 🗂 [Databases](databases.md)
```

File: databases.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+* [Sequel Pro (Mac)](https://www.sequelpro.com/)
+* [DataGrip (Universal)](https://www.jetbrains.com/datagrip/download/)
+* [MySQL Workbench (Universal)](https://www.mysql.com/products/workbench/)
+* [Realm Studio (Universal)](https://realm.io/products/realm-studio/)
+
+## Practices
+
+### Identitys
+
+ID's should be either auto-incrementing unsigned integers, or GUIDs/UUIDs.
```
Comment (dextermb) [inline databases.md]: Perhaps it's worth mentioning that in SQL you do this by creating an `IDENTITY` on a specific field

File: databases.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+
+### Approved Database Management Tools
+Below are a list of approved database management tools that you can download and use.
+
+* [Sequel Pro (Mac)](https://www.sequelpro.com/)
+* [DataGrip (Universal)](https://www.jetbrains.com/datagrip/download/)
+* [MySQL Workbench (Universal)](https://www.mysql.com/products/workbench/)
+* [Realm Studio (Universal)](https://realm.io/products/realm-studio/)
+
```
Comment (AlexCatch) [inline databases.md]: I use TablePlus 👍 

```suggestion
* [TablePlus](https://tableplus.io/ (Mac/ iOS))
```

**Comment by dextermb** [comment]
> Table Plus is what I use for db management so adding it to the approved list is a good idea.
> https://tableplus.io/

Probably a good alternative to Sequel Pro as I believe they stopped updating it properly (last proper update 2016).
I haven't used Table Plus myself though.

File: databases.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+Below are a list of approved database management tools that you can download and use.
+
+* [Sequel Pro (Mac)](https://www.sequelpro.com/)
+* [DataGrip (Universal)](https://www.jetbrains.com/datagrip/download/)
+* [MySQL Workbench (Universal)](https://www.mysql.com/products/workbench/)
+* [Realm Studio (Universal)](https://realm.io/products/realm-studio/)
+* [TablePlus (Mac/ iOS)](https://tableplus.io/)
+## Practices
+
+### Identitys
```
Comment (DivineOmega) [inline databases.md]:
```suggestion
### Identities
```

**Pull request merged**